usr/src/uts/armv8/boot/boot_console.c

```diff
@@ -0,0 +1,887 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+ * or http://www.opensolaris.org/os/licensing.
+ * See the License for the specific language governing permissions
+ * and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+/*
+ * Copyright (c) 2012 Gary Mills
+ * Copyright 2020 Joyent, Inc.
+ * Copyright 2025 Michael van der Westhuizen
+ *
+ * Copyright 2009 Sun Microsystems, Inc.  All rights reserved.
```

Comment:
Is the plan that we'll fold this back into common code as much as we can later, like we've tried to do with other code we've copied to work out what it ends up looking like?

Reply:
Most of boot_fb.c can be shared (the only difference is parsing a UEFI frame buffer instead of a multiboot2 frame buffer). The boot_console code has many sharable chunks which could be extracted to a common file, but the logic around selecting the console and diag device differs quite a lot, as done dispatching to the UART (though, again, the i86pc code could factor that out and we could make the names the same in the future).